File: 백준/Silver/2606. 바이러스/바이러스.cc
#include <iostream>
#
using namespace std;
int n;
int arr[101][101];
int visited[101];
int count = 0;
void dfs(int v) {
    visited[v] = 1;
    count++;
    for(int i=1;i<=n;i++) {
        if( arr[v][i] == 1 && visited[i] == 0) {
            dfs(i);
        }
        if(i==n)
            return ;
    }
}
int main() {
    cin>>n;
    int m;
    cin>>m;
    for(int i=0;i<m;i++) {
        int a,b;
        cin>>a>>b;
        arr[a][b] = 1;
        arr[b][a] = 1;
    }
    dfs(1);
    
    cout<<count -1 ;
    
}

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int m = Integer.parseInt(br.readLine());
        List<List<Integer>> arr = new ArrayList<>();
        
        for(int i=0; i<=n; i++) {
            arr.add(new ArrayList<>());
        }
        
        StringTokenizer st;
        for(int i=0; i<m; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            arr.get(a).add(b);
            arr.get(b).add(a);
        }
        
        int[] visited = new int[n+1];
        Queue<Integer> q = new ArrayDeque<>();
        
        visited[1] = 1;
        q.add(1);
        int result = -1;
        while(!q.isEmpty()) {
            int now = q.remove();
            result++;
            
            for(Integer i : arr.get(now)) {
                if(visited[i]!=1) {
                    q.add(i);
                    visited[i]=1;
                }
            }
        }
        System.out.println(result);
    }
}
        